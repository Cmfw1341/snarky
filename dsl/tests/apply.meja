let a = {
  let x = 1;
  let f = fun (x) => { x; };
  f x;
};

let b = fun (f : _ -> _) => {
  let i : _ = f 15;
  let j : _ = f i;
  j;
};

let c = {
  let f : _ = fun (x : 'a) => {x;};
  f f f;
};

let d = fun (f : 'b -> 'b, x : int) => {
  f 15;
  f x;
};
