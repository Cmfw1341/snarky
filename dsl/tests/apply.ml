let a =
  let x = 1 in
  let f x = x in
  f x

let b (f : int -> '_a28) =
  let (i : '_a28) = f 15 in
  let (j : '_a28) = f i in
  j

let c =
  let (f : '_a51 -> '_a51) = fun (x : '_a51) -> x in
  f f f

let d (f : int -> int) (x : int) = f 15 ; f x
