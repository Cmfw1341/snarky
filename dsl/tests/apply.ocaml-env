[@@@ocaml.doc "Types:"]
type bool =
  | true 
  | false 
type char
type float
type int
type string
type unit =
  | () 
[@@@ocaml.doc "Values:"]
val a : int
val b : (int -> int) -> int
val c : 'a -> 'a
val d : (int -> int) -> int -> int
[@@@ocaml.doc "Fields:"]
[@@@ocaml.doc "Constructors:"]
val () : unit
val false : bool
val true : bool