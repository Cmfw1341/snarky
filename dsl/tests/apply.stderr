Uncaught exception:
  
  (Failure "Could not find name x.")

